spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/XsuaaJwtDecoder.java

```diff
@@ -127,8 +126,7 @@ private Jwt verifyToken(JWT jwt) {
 	private Jwt verifyToken(String token, String jku, String kid, String uaaDomain) {
 		try {
 			canVerifyWithKey(jku, kid, uaaDomain);
-			validateJku(jku, uaaDomain);
-			return verifyWithKey(token, jku, kid);
+			return verifyWithKey(token, composeJku(uaaDomain), kid);
```

Comment:
We need to pass the zid from the token to composeJku and there, append it as query parameter to the URL, e.g. .../token_keys?zid=\<tokenZid\>.
If it is not in the token or blank, this query parameter can be omitted.